test/metabase/upload_test.clj

```diff
@@ -539,387 +598,372 @@
-            long-name    (apply str (repeat 33 "abcdefgh")) ; 33×8 = 264. Max is H2 at 256
-            short-name   (subs long-name 0 (- length-limit (count "_yyyyMMddHHmmss")))]
+            long-name    (apply str (shuffle (apply concat (repeat 33 (seq "abcdefgh"))))) ; 33×8 = 264. Max is H2 at 256
+            short-name   (subs long-name 0 (- length-limit (count "_yyyyMMddHHmmss")))
+            table-name   (u/upper-case-en (@#'upload/unique-table-name driver/*driver* long-name))]
         (is (pos? length-limit) "driver/table-name-length-limit has been set")
         (with-mysql-local-infile-on-and-off
           (mt/with-empty-db
             (@#'upload/load-from-csv!
              driver/*driver*
              (mt/id)
-             (@#'upload/unique-table-name driver/*driver* long-name)
+             table-name
              (csv-file-with ["number,bool"
                              "1,true"
                              "2,false"
                              (format "%d,true" Long/MAX_VALUE)]))
-            (testing "It truncates it to the right number of characters, allowing for the timestamp"
-              (sync/sync-database! (mt/db))
-              (let [table    (t2/select-one Table :db_id (mt/id) :%lower.name [:like (str short-name "%")])
-                    table-re (re-pattern (str "(?i)" short-name "_\\d{14}"))]
-                (is (re-matches table-re (:name table)))
-                (testing "Check the data was uploaded into the table correctly"
-                  (is (= [[1 1 true]
-                          [2 2 false]
-                          [3 Long/MAX_VALUE true]]
-                         (rows-for-table table))))))))))))
+            (let [table    (sync-upload-test-table! :database (mt/db) :table-name table-name)
+                  table-re (re-pattern (str "(?i)" short-name "_\\d{14}"))]
+              (testing "It truncates it to the right number of characters, allowing for the timestamp"
+                (is (re-matches table-re (:name table))))
+              (testing "Check the data was uploaded into the table correctly"
+                (is (= [[1 1 true]
+                        [2 2 false]
+                        [3 Long/MAX_VALUE true]]
+                       (rows-for-table table)))))))))))
 
 (deftest load-from-csv-empty-header-test
   (testing "Upload a CSV file with a blank column name"
     (mt/test-drivers (mt/normal-drivers-with-feature :uploads)
       (mt/with-empty-db
-        (@#'upload/load-from-csv!
-         driver/*driver*
-         (mt/id)
-         "upload_test"
-         (csv-file-with [",ship name,"
-                         "1,Serenity,Malcolm Reynolds"
-                         "2,Millennium Falcon, Han Solo"]))
-        (testing "Table and Fields exist after sync"
-          (sync/sync-database! (mt/db))
-          (let [table (t2/select-one Table :db_id (mt/id))]
-            (is (=? {:name #"(?i)upload_test"} table))
```

Comment:
We did we remove this assertion instead of updating it? I guess it was being repeated a lot, but is it still tested at least once somewhere?